training/training/core/dataset.py

```diff
@@ -201,3 +203,80 @@ def createTestDataset(self) -> DataLoader:
 
     def getCategoryList(self) -> list[str]:
         return self.train_set.classes if hasattr(self.train_set, "classes") else []
+
+class UrbanSoundDataset(Dataset):
+    
+    def __init__(self, dataset, folderList):
+        # Initialize lists to hold file names, labels, and folder numbers
+        self.file_names = []
+        self.labels = []
+        self.folders = []
+        self.dataset = dataset
+        
+        # Loop through the dataset and only add entries from folders in the folder list
+        for clip in dataset.clip_ids:
+            clip_metadata = dataset.clip(clip)
+            fold = int(clip_metadata.subset.split('_')[1])  # Extract fold number
+            if fold in folderList:
+                self.file_names.append(clip_metadata.audio_path)
+                self.labels.append(clip_metadata.tags[0])  # Assuming the first tag is the label
+                self.folders.append(fold)
+        
+        self.folderList = folderList
+        
+    def __getitem__(self, index):
+        # Load the file
+        path = self.file_names[index]
+        sound, sample_rate = torchaudio.load(path)
+        
+        # Convert to mono if stereo. Otherwise, just take the first channel. 
+        # we need this bc the UrbanSound8k dataset is stereo, but torchaudio defaults to mono
+        if sound.size(0) > 1:
+            sound = torch.mean(sound, dim=0, keepdim=True)
+        
+        # Normalize the audio. Mean and std are calculated from the training set
+        sound = (sound - sound.mean()) / sound.std()
+
+        # Downsample the audio to ~8kHz
```

Comment:
did the paper do these preprocessing steps?

Reply:
yes. this is just simple normalization.

Reply:
got it. maybe we should have a utility preprocess function that stores these steps so that anyone who wants to learn more can better follow the logic

Reply:
by this I mean a helper preprocess function for this specific dataset class you created